PresentScreenings/Entities/ScreeningsPlan.cs

```diff
@@ -130,12 +145,52 @@ select s
                 }
             }
         }
+
+        public Screen DisplayScreen(Screening screening)
+        {
+            if (screening is OnDemandScreening onDemandScreening)
+            {
+                return onDemandScreening.DisplayScreen;
+            }
+            return screening.Screen;
+        }
+
+        private void AssignDisplayScreens()
+        {
+            var screenRegex = new Regex(@"\D+(\d*)");
+            var displayScreenByAbbreviation = new Dictionary<string, Screen> { };
+
+            foreach (var screening in Screenings)
+            {
+                if (screening is OnDemandScreening onDemandScreening)
+                {
+                    int filmId = onDemandScreening.FilmId;
+                    int screenId = onDemandScreening.Screen.ScreenId;
+                    string odAbbreviation = onDemandScreening.Screen.Abbreviation;
+                    Screening LocalScreening = Screenings
+                        .Where(s => s.FilmId == filmId && s.Screen.ScreenId != screenId)
+                        .First();
+                    Match match = screenRegex.Match(LocalScreening.Screen.Abbreviation);
+                    if (match != null)
+                    {
+                        string version = match.Groups[1].Value;
+                        string newAbbreviation = odAbbreviation + version;
+                        if (!displayScreenByAbbreviation.ContainsKey(newAbbreviation))
+                        {
+                            displayScreenByAbbreviation.Add(newAbbreviation, new Screen(onDemandScreening.DisplayScreen, newAbbreviation));
+                        }
+                        onDemandScreening.DisplayScreen = displayScreenByAbbreviation[newAbbreviation];
+                    }
+                    onDemandScreening.SetEndTime(LocalScreening);
```

Comment:
Please explain this functionally unrelated statement in a comment.

Reply:
Added several comments concerning initialization of specific on-demands features.